compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -1137,9 +1133,8 @@ impl<'a, 'tcx> CrateMetadataRef<'a> {
                                 // within the crate. We only need this for fictive constructors,
                                 // for other constructors correct visibilities
                                 // were already encoded in metadata.
-                                let attrs: Vec<_> =
-                                    self.get_item_attrs(def_id.index, sess).collect();
-                                if sess.contains_name(&attrs, sym::non_exhaustive) {
+                                let mut attrs = self.get_item_attrs(def_id.index, sess);
+                                if attrs.any(|item| sess.check_name(&item, sym::non_exhaustive)) {
```

Comment:
You should be able to use cheaper `fn has_name` here, I don't think we are reporting unused attribute warnings for other crates.

Reply:
Good point, thanks